box_types_metadata.go

```diff
@@ -167,6 +175,85 @@ func (sd *StringData) StringifyField(name string, indent string, depth int, ctx
 	return "", false
 }
 
+// Item is a item under an item list
+// https://developer.apple.com/documentation/quicktime-file-format/metadata_item_list_atom/item_list
+type Item struct {
+	AnyTypeBox
+	Version  uint8   `mp4:"0,size=8"`
+	Flags    [3]byte `mp4:"1,size=8"`
+	ItemName []byte  `mp4:"2,size=8,len=4"`
+	Data     Data    `mp4:"3"`
```

Comment:
💭 The reason `Item` here is of `AnyBoxType` is because `AddAnyTypeBoxDefEx()` requires the BoxType be of `AnyTypeBox`. 

I'm curious if there is a better approach to registering `Data` field under a number `Item` BoxType? The current implementation of having `Data` be a field under the `Item` AnyTypeBox doesn't look entirely clean. I think it'd be better if the `Data` BoxType here is handled as its own atom but it may require changing/adding a func similar to `AddAnyTypeBoxDefEx()` but that takes in a `FullBox` so that we only discovered number atom blocks if they are under the `ilst` atom?